yt/visualization/base_plot_types.py

```diff
@@ -334,10 +343,12 @@ def _init_image(self, data, cbnorm, cblinthresh, cmap, extent, aspect):
                         10
                         ** np.arange(
                             np.rint(np.log10(cblinthresh)),
-                            np.ceil(np.log10(zmax)),
+                            np.ceil(np.log10(1.1 * zmax)),
                         )
                     )
                 )
+            if yticks[-1] > zmax:
+                yticks.pop()
```

Comment:
necessary cleanup with matplotlib 3.5 where having a ticks that's above the actual maximal value creates a white region at the top end of the colorbar (see https://github.com/yt-project/yt/issues/3554)